src/components/common/UserProfile/UserProfile.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+
+import ProfileTemplate from '~frame/ProfileTemplate';
+import NotFoundRoute from '~routes/NotFoundRoute';
+import { useUserByNameOrAddress } from '~hooks';
+
+import UserMeta from './UserMeta';
+import UserProfileSpinner from './UserProfileSpinner';
+import UserColonies from './UserColonies';
+
+import styles from './UserProfile.css';
+
+const displayName = 'common.UserProfile';
+
+const UserProfile = () => {
+  const { usernameOrAddress } = useParams();
+  const { user, loading, error } = useUserByNameOrAddress(
+    usernameOrAddress || '',
+  );
+
+  if (loading) {
+    return <UserProfileSpinner />;
+  }
+
+  if (error || !user) {
+    return <NotFoundRoute />;
+  }
```

Comment:
Does this need to implement the user `undefined` / `null` logic at all ?

Reply:
This one doesn't as it doesn't matter if we are logged in or not.